### PR TITLE
Fix #71 RSPEC-126 if else if without ending else

### LINKED ISSUE #71: Rule S126: "if ... else if" constructs should end with "else" clauses
Implements [RSPEC-126](https://jira.sonarsource.com/browse/RSPEC-126)

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/IfElseIfWithoutElseCheck.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    super.enterFile(inputFile);
+  }
+
+  @Override
+  public void visitNode(UastNode node) {
+    if(visitedIf.contains(node)) {
+      return;
+    }
+    IfLike ifLike = IfLike.from(node);
+    IfLike.ElseLike elseLike = ifLike.elseLike();
```
Comment (SonarTech) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/IfElseIfWithoutElseCheck.java]: Bug Bug: A "NullPointerException" could be thrown; "ifLike" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.go%3Asonar-go&pullRequest=277&issues=AWL4MyLYqJ_F9Z_shBXq&open=AWL4MyLYqJ_F9Z_shBXq)

[ISSUE_KEY]: AWL4MyLYqJ_F9Z_shBXq

File: its/ruling/src/test/expected/go-S126.json [hunk 2]
```diff
@@ -0,0 +1,2682 @@
...
+117,
+],
+'project:kubernetes/pkg/api/v1/resource/helpers.go':[
+94,
+],
+'project:kubernetes/pkg/api/v1/validation/validation.go':[
+50,
+],
+'project:kubernetes/pkg/api/validation/validation.go':[
+1551,
```
Comment (saberduck) [inline its/ruling/src/test/expected/go-S126.json:66]: Maybe we could exclude cases where `else` would be empty?

**Review by saberduck — COMMENTED** [review]
@benzonico have a look at my comment, if you want to consider it or not.

File: its/ruling/src/test/expected/go-S126.json (hunk 2 above)
Comment (benzonico) [reply its/ruling/src/test/expected/go-S126.json:66]: I am not sure I understand your suggestion,
given that RSPEC states : 
> The else statement should either take appropriate action or contain a suitable comment as to why no action is taken.

Comment (saberduck) [reply its/ruling/src/test/expected/go-S126.json:66]: OK, nevermind

**Pull request merged**